libcxx/include/string

```diff
@@ -1959,10 +1960,10 @@ private:
     if (__s < __min_cap) {
       return static_cast<size_type>(__min_cap) - 1;
     }
-    const size_type __boundary = sizeof(value_type) < __alignment ? __alignment / sizeof(value_type) : 1;
+    const size_type __boundary = sizeof(value_type) < __alignment ? __alignment / sizeof(value_type) : __endian_factor;
     size_type __guess          = __align_it<__boundary>(__s + 1) - 1;
     if (__guess == __min_cap)
-      ++__guess;
+      __guess += 2;
```

Comment:
works with ASSERT, and likely will work with asan
Should `__guess += 2` be replaced with  `__guess += __endian_factor` if we want to keep  __endian_factor alignment?

Reply:
> works with ASSERT, and likely will work with asan Should `__guess += 2` be replaced with `__guess += __endian_factor` if we want to keep __endian_factor alignment?

Assuming __endian_factor will be `>2` is unlikely, 2 will work, but __endian_factor is better for reader?

Reply:
Yeah, I think `__endian_factor` would be the right call. Maybe we should rename it into something like `__min_capacity_alignment`, since that's what it basically is? Just a though, not a request for this patch.

Reply:
Done